sklearn/datasets/kddcup99.py

```diff
@@ -11,32 +11,35 @@
 import sys
 import errno
 from gzip import GzipFile
-from io import BytesIO
 import logging
 import os
 from os.path import exists, join
-try:
-    from urllib2 import urlopen
-except ImportError:
-    from urllib.request import urlopen
 
 import numpy as np
 
+
+from .base import _fetch_url
 from .base import get_data_home
+from .base import RemoteFileMetadata
 from ..utils import Bunch
 from ..externals import joblib, six
 from ..utils import check_random_state
 from ..utils import shuffle as shuffle_method
 
 
-URL10 = ('http://archive.ics.uci.edu/ml/'
-         'machine-learning-databases/kddcup99-mld/kddcup.data_10_percent.gz')
-
-URL = ('http://archive.ics.uci.edu/ml/'
-       'machine-learning-databases/kddcup99-mld/kddcup.data.gz')
+ARCHIVE = RemoteFileMetadata(
```

Comment:
It is at the [beginning of the file](https://github.com/scikit-learn/scikit-learn/pull/9240/files/34fed87be14dfd930404f61c2c830af880e85757#diff-bf281b3176e3a8090f0ec9f2ebca82e2R7)